cpp/memilio/utils/custom_index_array.h

```diff
@@ -189,7 +189,9 @@ class CustomIndexArray
      * @param b begin of the range of values.
      * @param e end of the range of values.
      */
-    template <class Iter>
+    template <class Iter,
+              typename std::enable_if_t<std::is_base_of<
+                  std::input_iterator_tag, typename std::iterator_traits<Iter>::iterator_category>::value>* = nullptr>
     CustomIndexArray(Index const& dims, Iter b, Iter e)
```

Comment:
Yes, otherwise it chooses this constructor for arrays that receive two arguments after the dimensions.

I guess the `is_base_of` is not necessary/wrong. I have implemented a type trait in `stl_util.h` and use that here. Please comment for improvements.

Reply:
is_base_of is fine. I think the problem with your approach is 
- readability
- I think, if iterator_traits cannot be instantiated for the type for some reason, that's a hard error instead of nice SFINAE.

Reply:
no, the way you changed it is fine. I was still talking about the first approach.